File: ipc_read_shared_memory.py
import sysv_ipc
import warnings
SHM_SIZE = 1024
# Ignore warnings from sysv_ipc library
with warnings.catch_warnings():
warnings.simplefilter("ignore")
key = sysv_ipc.ftok("shmfile", 65)
# Attach to the shared memory segment
shm = sysv_ipc.SharedMemory(key)
# Read the message from the shared memory segment
message = shm.read().decode('utf-8').rstrip('\0')
print(f"Message from shared memory: {message}")
# Detach from the shared memory segment
shm.detach()
# Remove the shared memory segment
shm.remove()

File: ipc_read_shared_memory.c
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/ipc.h>
#include <sys/shm.h>
#include <sys/types.h>
#include <unistd.h>
#define SHM_SIZE 1024
int main() {
int shmid;
key_t key;
char *shm, *s;
key = ftok("shmfile", 65);
shmid = shmget(key, SHM_SIZE, 0666 | IPC_CREAT);
shm = shmat(shmid, NULL, 0);
// Read the message from the shared memory segment
printf("Message from shared memory: %s\n", shm);
shmdt(shm);
shmctl(shmid, IPC_RMID, NULL);
return 0;
}
